pkg/plugins/manager/process/process.go

```diff
@@ -54,12 +54,13 @@ func (m *Manager) Start(ctx context.Context, pluginID string) error {
 
 	m.log.Info("Plugin registered", "pluginID", p.ID)
 	m.mu.Lock()
-	if err := startPluginAndRestartKilledProcesses(ctx, p); err != nil {
+	err := startPluginAndRestartKilledProcesses(ctx, p)
+	m.mu.Unlock()
+	if err != nil {
```

Comment:
It's more idiomatic to `defer` the Unlock:

```suggestion
	defer m.mu.Unlock()
	if err := startPluginAndRestartKilledProcesses(ctx, p); err != nil {
```